server/gpu/modal_deployments/reflector_transcriber.py

```diff
@@ -130,32 +141,106 @@ def web():
-        if apikey != os.environ["REFLECTOR_GPU_APIKEY"]:
-            raise HTTPException(
-                status_code=status.HTTP_401_UNAUTHORIZED,
-                detail="Invalid API key",
-                headers={"WWW-Authenticate": "Bearer"},
-            )
+        if apikey == os.environ["REFLECTOR_GPU_APIKEY"]:
+            return
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid API key",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
 
-    class TranscriptResponse(BaseModel):
-        result: dict
+    def detect_audio_format(url: str, headers: dict) -> str:
+        from urllib.parse import urlparse
+
+        url_path = urlparse(url).path
+        for ext in supported_file_types:
+            if url_path.lower().endswith(f".{ext}"):
+                return ext
+
+        content_type = headers.get("content-type", "").lower()
+        if "audio/mpeg" in content_type or "audio/mp3" in content_type:
+            return "mp3"
+        if "audio/wav" in content_type:
+            return "wav"
+        if "audio/mp4" in content_type:
+            return "mp4"
+
+        raise HTTPException(
+            status_code=400,
+            detail=(
+                f"Unsupported audio format for URL. Supported extensions: {', '.join(supported_file_types)}"
+            ),
+        )
 
     @app.post("/v1/audio/transcriptions", dependencies=[Depends(apikey_auth)])
     def transcribe(
-        file: UploadFile,
-        model: str = "whisper-1",
-        language: Annotated[str, Body(...)] = "en",
-    ) -> TranscriptResponse:
-        audio_data = file.file.read()
-        audio_suffix = file.filename.split(".")[-1]
-        assert audio_suffix in supported_file_types
+        file: UploadFile | None = None,
+        files: list[UploadFile] | None = None,
+        model: str = Form(MODEL_NAME),
+        language: str = Form("en"),
+        batch: bool = Form(False),
+    ):
+        if not file and not files:
+            raise HTTPException(
+                status_code=400, detail="Either 'file' or 'files' parameter is required"
+            )
+
+        upload_files = [file] if file else files
+        results = []
+
+        for upload_file in upload_files:
+            audio_suffix = upload_file.filename.split(".")[-1]
+            assert audio_suffix in supported_file_types
+
+            content = upload_file.file.read()
+            func = transcriber.transcribe_segment.spawn(
+                audio_data=content,
+                audio_suffix=audio_suffix,
+                language=language,
+            )
+            result = func.get()
+            result["filename"] = upload_file.filename or str(uuid.uuid4())
+            results.append(result)
+
+        return {"results": results} if len(results) > 1 else results[0]
+
+    @app.post("/v1/audio/transcriptions-from-url", dependencies=[Depends(apikey_auth)])
```

Comment:
You implemented this one, but using the same transcribe segment. This is going to fail, you can't pass 1h meeting and ask for transcription, there won't be enough GPU memory right ? Or does whisper already do everything by itself ?